### DISCUSSION TITLE
logging into formR from survey - secure?

### DISCUSSION THREAD

**Discussion by kiraschweickhardt** [rubenarslan/formr.org#583] [Q&A] [answered]
Hi all,

I want to access results from different runs so I can calculate quotas. I found this discussion with a solution for this:

https://github.com/rubenarslan/formr.org/discussions/519

However, to apply this, I would have to write my formR password and login-name in the value column of a calculate item and integrate that survey into my run. I know, that calculate items are of the „hidden“  item family. Still, I was wondering: Is it sure that my login-information can’t be accessed by someone who would be filling out that survey?

Thank you in advance!
Kira

**Comment by rubenarslan** [comment, marked as answer]
Hi Kira,

I don't recommend using `formr_connect` that way. Users should never see the R code you entered, but it's still bad practice to store your own password in plaintext in the database and on OpenCPU. If an attacker got access to the DB, they could log into your account. We have a proper API which allows the creation of access tokens, but this is not well documented at the moment.
In the newer versions of formr (in preparation for Münster atm), there is a variable called "iteration" in every survey, a number which increases, the more people take part. I am not sure whether you could solve your problem this way. You would probably have to move to a single run. Generally, the idea in formr is for a run to be self-contained, although there are of course exceptions where multiple runs solve your problem better.
